src/librustc_mir/build/expr/as_place.rs

```diff
@@ -28,22 +28,38 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
         where M: Mirror<'tcx, Output=Expr<'tcx>>
     {
         let expr = self.hir.mirror(expr);
-        self.expr_as_place(block, expr)
+        self.expr_as_place(block, expr, Mutability::Mut)
+    }
+
+    /// Compile `expr`, yielding a place that we can move from etc.
```

Comment:
Nit: I'd add a comment here like:

Mutability note: The caller of this method promises only to read from the resulting
place. The place itself may or may not be mutable: 
- If this expr is a place expr like `a.b`, then we will return that place.
- Otherwise, a temporary is created: in that event, it will be an immutable temporary.

Reply:
Maybe argues for calling it `as_read_only_place`